### PR TITLE
Gold data generation for wavelet-to-realspace conversion.

### PR DESCRIPTION
**Pull request by stonecoldhughes**
Gold data generation for matrix_plot_D
Generated gold data for both routines for continuity1,2,3

### CODE REVIEW

**Comment by bmcdanie** [comment]
`make_all_gold` should generate all gold files.

currently:

```
Not enough input arguments.

Error in gold_matrix_plot_D (line 4)
data_dir = strcat("generated-inputs", "/", "matrix_plot_D", "/", output_prefix, "/");

Error in make_all_gold (line 35)
gold_matrix_plot_D
 
```
Are we missing generation of wavelet to realspace transform golds from 'make_all`?

Final note - the `gold_` scripts should generate golds for all sizes/configurations tested like the others, and store the results in a folder named for the component under test.

**Comment by stonecoldhughes** [comment]
ah, I made these 2 gold scripts functions so that I could quickly run many different problem configurations after 6d blew everything up. I'll go back and fix this

File: gold/gold_matrix_plot_D.m [hunk 1]
```diff
@@ -1,15 +1,22 @@
-%% Generate gold data for C++ testing of matrix_plot_D component
-data_dir = strcat("generated-inputs", "/", "matrix_plot_D", "/");
+gen_realspace_transform( continuity1, 'continuity_1', 'lev', 8, 'deg', 7);
```
Comment (dlg0) [inline gold/gold_matrix_plot_D.m:1]: I'm unfamiliar with this `gen_realspace_transform` function? Where is it? Also, I'm not sure why this PR needs to modify any other gold data generation routine other than the new `gold_wavelet_to_realspace`?

Comment (bmcdanie) [reply gold/gold_matrix_plot_D.m:1]: Gen realspace transform is the c++ equivalent of matrix plot D. This function is called from wavelet to realspace and is tested separately.

I actually pulled these changes into my pr, but you can merge them and I'll rebase if you'd prefer to keep them separate.

Comment (dlg0) [reply gold/gold_matrix_plot_D.m:1]: @bmcdanie which PR are you referring to? Are you suggesting these changes are already merged in?

Comment (bmcdanie) [reply gold/gold_matrix_plot_D.m:1]: yes - I believe they are. this one had been sitting a while, so I updated the file to produce the gold values we test on and added it to a previous MR.

Comment (stonecoldhughes) [reply gold/gold_matrix_plot_D.m:1]: This PR needed to modify 3 gold data generation routines I believe - 
1. gold_wavelet_to_realspace
2. gold_matrix_plot_D
3. make_all_gold

The changes in the remaining 6 files consist of added whitespace for readability, removal of trailing hidden characters at the ends of lines, and and autogenerated file in the two_scale directory

**Pull request merged**